tensorflow-core/tensorflow-core-api/src/main/java/org/tensorflow/device/DeviceSpec.java

```diff
@@ -0,0 +1,166 @@
+package org.tensorflow.device;
+
+import com.google.common.base.Joiner;
```

Comment:
Sorry @hthu , you should not use Guava neither :)

Guava is present in the classpath at compile time only as a result of depending on `tensorflow-core-generator` for running the annotation processor. As you can [see in the pom](https://github.com/tensorflow/java/blob/81a2b605da90689fbdaed975106fb2bd5f85ca6c/tensorflow-core/tensorflow-core-api/pom.xml#L41), this dependency is optional, meaning that it will not be automatically present at runtime nor its transitive dependencies. I'm even surprised they were included when running your unit tests.

Again, sorry for the trouble but I think we should try to avoid adding a new runtime dependency in the core module if there is an easy way to accomplish the same task without it.

Reply:
NP!
Removed as well.